### PR TITLE
Add plain text lossless export option

### LINKED ISSUE #991: Export: Plain Text (lossless)
Add an export type of "Plain Text (lossless)" with all metaprogramming attributes included. 

Change the default export type of "Plain Text" to exclude all metaprogramming attributes.

### PR DESCRIPTION
**Pull request by shresthabijay**
fixes #991 

# Changes
- Add plain text lossless and without loss export option.
- Refactor dropdown ui.
- Add on click outside to close dropdown.

### CODE REVIEW

**Comment by shresthabijay** [comment]
@raineorshine  Please review the changes. Thanks!

File: src/util/filter.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+interface Predicate<P> {
+  (item: P, index: number): boolean,
+}
+
+/**
+ * Filter with multiple predicate functions.
+ */
+export const filter = <T>(array: T[], predicateArray: (Predicate<T>)[]) => {
+  return predicateArray.length > 0 ? array.filter((item: T, index) => predicateArray.every(predicate => predicate(item, index))) : array
+}
```
Comment (raineorshine) [inline src/util/filter.ts:10]: This functionality can be achieved more simply with function composition:

```
items.filter(and(predicate1, predicate2, etc))
```

`and` is a basic higher order function. In order to get access to it you can import [fp-and-or](https://github.com/raineorshine/fp-and-or).

**Review by raineorshine — CHANGES_REQUESTED** [review]
Great! Works as expected. Just one change requested.

File: src/util/filter.ts (hunk 1 above)
Comment (shresthabijay) [reply src/util/filter.ts]: That makes sense. Thank you!

**Comment by shresthabijay** [comment]
@raineorshine  I have made the change.

File: src/util/filter.ts (hunk 1 above)
Comment (raineorshine) [reply src/util/filter.ts]: Yeah, super handy. I'm surprised they're not in lodash. I finally made a package just for them since I use them so often :).

**Review by raineorshine — APPROVED** [review]

**Pull request merged**